test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Rendering/HtmlHelperTextBoxTest.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Mvc.TestCommon;
+using System.ComponentModel.DataAnnotations;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Rendering
+{
+    public class HtmlHelperTextBoxTest
```

Comment:
@tuespetre not :100:% required: Please add similar test methods to `InputTagHelperTest`? It uses the same code paths through `DefaultHtmlGenerator` and passes in an `htmlAttributes` argument that includes the `type` which will really be generated. Definitely do not need to double-check _all_ of these values.